### PR TITLE
React/postmessage interface

### PR DESCRIPTION
**Pull request by joshkadis**
Update how data is handled and saved by plugin to match the new data structure.

### CODE REVIEW

File: js/plugin/post-edit.js [hunk 1]
```diff
@@ -1,170 +1,144 @@
...
+		window.addEventListener('message', onReceiveMessage );
+		renderOpenModal();
+	}
 
-		clearInputEl : function(e) {
+	/**
+	 * Renders the app iframe modal in its open state
+	 */
+	function renderOpenModal() {
+			// create modal elements and append to <body>
```
Comment (scottnelle) [inline js/plugin/post-edit.js]: Extra indent here.

File: js/plugin/post-edit.js [hunk 2]
```diff
@@ -1,170 +1,144 @@
...
-			// store published chart ID
-			if ( ! _.isUndefined( app.chartData.id ) ) {
-				app.inputChartIdEl.value = app.chartData.id;
-			}
+	/**
+	 * Extract messageType string when a postMessage is received
+	 */
+	function getMessageType(evt) {
+		// confirm same-origin or http(s)://localhost:8080
+		if ( evt.origin !== window.location.origin && !/https?:\/\/localhost:8080/.test(evt.origin )) {
```
Comment (scottnelle) [inline js/plugin/post-edit.js]: inconsistent spacing inside parentheses for this conditional and in the call to the `test()` method. I believe our JS practice is no spaces, but either way it should be consistent.

File: modules/class-simplechart-save.php [hunk 3]
```diff
@@ -91,30 +98,14 @@ function do_save_post( $post ) {
 		}
 
 		// handle base64 image string if provided
-		if ( ! empty( $_POST['simplechart-png-string'] ) ) {
-			$this->_save_chart_image( $post, $_POST['simplechart-png-string'], $this->_default_img_type );
+		if ( ! empty( $_POST['save-previewImg'] ) ) {
+			$this->_save_chart_image( $post, $_POST['save-previewImg'], $this->_default_img_type );
```
Comment (scottnelle) [inline modules/class-simplechart-save.php]: Doesn't look like `$_POST['save-previewImg']` ever gets sanitized.

File: modules/class-simplechart-save.php [hunk 4]
```diff
@@ -91,30 +98,14 @@ function do_save_post( $post ) {
...
-		if ( ! empty( $_POST['simplechart-chart-url'] ) ) {
-			update_post_meta( $post->ID, 'simplechart-chart-url',  esc_url( $_POST['simplechart-chart-url'] ) );
-		}
-
-		// save chart ID if provided
-		if ( ! empty( $_POST['simplechart-chart-id'] ) ) {
-			update_post_meta( $post->ID, 'simplechart-chart-id',  sanitize_text_field( $_POST['simplechart-chart-id'] ) );
+		foreach ( $this->meta_field_names as $field ) {
+			if ( ! empty( $_POST[ 'save-' . $field ] ) ) {
+				update_post_meta( $post->ID, 'save-' . $field, $_POST[ 'save-' . $field ] );
```
Comment (scottnelle) [inline modules/class-simplechart-save.php]: Sanitize.

**Comment by scottnelle** [comment]
A couple small things. 👻

File: modules/class-simplechart-save.php (hunk 4 above)
Comment (joshkadis) [inline modules/class-simplechart-save.php]: Doesn't `update_post_meta()` handle sanitizing?

Comment (scottnelle) [inline modules/class-simplechart-save.php]: Maybe? It passes to `update_metadata()` which passes to `sanitize_meta()` which applies filters based on the meta type and meta key. It further runs through some filters for the meta type. You can always see what VIP has to say. I suspect they'll ask you to sanitize early.

**Pull request merged**